std_traffic/pipelines/sumoToAws.sh

```diff
@@ -330,40 +368,67 @@ load_db() {
     export DB_USER=${DB_USER}
     # find all files generated by awk
     # and load them in PostgreSQL
-    find "$(pwd -P)" $TMP_DIR -maxdepth 1 -name "*${SUMO_OUTPUT_FILE}.csv.0*" -exec \
-        load_csv_psql.py {} ${DATABASE} ${TABLE} \
-        --indexes "${DB_INDEXES}" \;
+    # https://unix.stackexchange.com/a/392973 modified to have more argument
+    # NOTE. -exec + will try to put as many filenames as possible in a single run
+    # then our code iterates over {@:3}-- whatevere was put therein
+    find "$(pwd -P)" $TMP_DIR -maxdepth 1 -name "*${SUMO_OUTPUT_FILE}.csv.0*" \
+        -exec /bin/bash -c 'for n in "${@:3}"; do load_csv_psql.py "$n" "$0" "$1" --indexes "$2" || exit 1; done' \
+        ${DATABASE} ${TABLE} "${DB_INDEXES}" "{}" +
 }
 
 
 # load data into AWS/RDS
 if [ $DB -eq 1 ]
 then
-    logit "Starting CSV file split..."
-    if split_csv
+    if [ $CSV_DONE -eq 0 ] && [ $PY_OK -eq 0 ]
     then
-        logit "CSV split completed"
-        if load_db
+        logit "Starting CSV file split..."
+        if split_csv
         then
-            logit "DB load completed"
+            logit "CSV split completed"
+            if load_db
+            then
+                logit "DB load completed"
+            else
+                DB_DONE=1
+                logit "DB load failed"
+            fi
         else
-            logit "DB load failed"
-            logit "Files not deleted in ${TMP_DIR}"
-            CLEANUP=0
+            logit "CSV split failed"
+            logit "Skipped Database load"
+            CSV_SPLIT_DONE=1
+            DB_DONE=1
         fi
     else
-        logit "CSV split failed"
-        logit "Files not deleted in ${TMP_DIR}"
-        CLEANUP=0
+        logit "Cannot load DB"
+        DB_DONE=1
+        CSV_SPLIT_DONE=1
     fi
+else
+    logit "DB load not requested"
 fi
 
 
 if [ $CLEANUP -eq 1 ]
 then
     logit "Cleaning up..."
-    rm $TMP_DIR/${SUMO_MODEL_PREFIX}.${SUMO_OUTPUT_FILE}*
+    if [ $SUMO_DONE -eq 0 ]
+    then
+        rm $TMP_DIR/${SUMO_MODEL_PREFIX}.${SUMO_OUTPUT_FILE}.xml*
+    fi
+
+    # delete the CSV only if everything else was OK
+    if [ $CSV_DONE -eq 0 ] && [ $S3_DONE -eq 0 ] && [ $DB_DONE -eq 0 ]
+    then
+        rm $TMP_DIR/${SUMO_MODEL_PREFIX}.${SUMO_OUTPUT_FILE}.csv
+    else
+        logit "Exiting with error. CSV saved at ${TMP_DIR}/${SUMO_MODEL_PREFIX}.${SUMO_OUTPUT_FILE}.csv"
+    fi
+
+    if [ $CSV_SPLIT_DONE -eq 0 ]
+    then rm $TMP_DIR/${SUMO_MODEL_PREFIX}.${SUMO_OUTPUT_FILE}.csv.0*
+    fi
 fi
 
-logit "All done"
+logit "Done."
```

Comment:
Don't know if I got it right, but here you're basically continuing all the way down to the end of the script even if a problem arised and completing the log even if doing anything, instead of just stopping the execution? If so, why?

Reply:
@RuggeroFabbiano You did, but this part went through a lot of changes because I wasn't sure what files make more sense to delete in case of failure, and which one not.
Sure we could stop anytime and just `exit 1`, and add some `rm` there, but after all I thought it was OK to have the cleanup part at the bottom.
How would you change it?

Reply:
On second thought yes, I guess it's neater that way.